### PR TITLE
Document Local K8s Setup

### PR DESCRIPTION
**Pull request by SeanCheatham**
## Purpose
- While the official runs of the Testnet Simulations will be in GCP, development of the tooling is aided by running the testnets locally
## Approach
- Document the installation steps for a local Kubernetes cluster using microk8s
## Tickets
- #2594

### CODE REVIEW

**Review by nandotorterolo — APPROVED** [review]

File: docs/LocalTestnetSimulatorGuide.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Local Testnet Simulator Guide
+Instructions for running a testnet simulation in a local environment.
+
+## Installation
+### Docker, Java, and SBT
+1. Install [Docker](https://docs.docker.com/get-docker/).
```
[the remark this answers is not in the document]
Comment (SeanCheatham) [reply docs/LocalTestnetSimulatorGuide.md]: Updated

File: docs/LocalTestnetSimulatorGuide.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+1. Install [Java](https://adoptium.net/installation/).
+1. Install [SBT](https://www.scala-sbt.org/download.html).
+
+NOTE: Java and SBT can be installed using [SDKMAN](https://sdkman.io/install)
+1. Run `curl -s "https://get.sdkman.io" | bash`
```
[the remark this answers is not in the document]
Comment (SeanCheatham) [reply docs/LocalTestnetSimulatorGuide.md]: Updated

File: docs/LocalTestnetSimulatorGuide.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+1. Run `source "$HOME/.sdkman/bin/sdkman-init.sh"`
+1. Run `sdk install java 11.0.17-tem`
```
[the remark this answers is not in the document]
Comment (SeanCheatham) [reply docs/LocalTestnetSimulatorGuide.md]: I don't have a good solution for this, other than to centralize the installation steps in _only_ one place in the repository.  I created a separate `DevelopmentSetup.md` and I've referenced it here in this document.

File: docs/LocalTestnetSimulatorGuide.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+1. Run `sdk install sbt 1.7.3`
+
+### Kubernetes
+1. Install [Microk8s](https://microk8s.io/docs/install-alternatives).
+1. Enable addons with command `microk8s enable dns helm helm3 hostpath-storage registry storage`
+
+### Bifrost
+1. Clone https://github.com/Topl/Bifrost and checkout branch `tetra`.
+1. Run `sbt "Docker / publishLocal"` from the command line.
```
[the remark this answers is not in the document]
Comment (SeanCheatham) [reply docs/LocalTestnetSimulatorGuide.md:14]: Updated

File: docs/LocalTestnetSimulatorGuide.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+1. Run `docker tag toplprotocol/network-delayer localhost:32000/topl/network-delayer && docker push localhost:32000/topl/network-delayer`
+1. Run `docker tag toplprotocol/bifrost-node-tetra localhost:32000/topl/bifrost-node-tetra && docker push localhost:32000/topl/bifrost-node-tetra`
+1. Run `docker tag toplprotocol/testnet-simulation-orchestrator localhost:32000/topl/testnet-simulation-orchestrator && docker push localhost:32000/topl/testnet-simulation-orchestrator`.
+
+## Run Simulation
+1. Open terminal.  `cd {bifrost repository}/helm`
+1. Copy `bifrost-consensus-testnet/override-example.yaml` to a path on your machine.  Update values accordingly.
+1. From terminal, run `microk8s helm install -f {path to override.yaml} --create-namespace --namespace {scenario name} {scenario name} ./bifrost-consensus-testnet`
+1. Wait for the test run to complete.  You can observe its progress by accessing the logs of the running pods.  [k9s](https://k9scli.io/) is a helpful tool but requires [kubeconfig setup](https://microk8s.io/docs/working-with-kubectl).
```
[the remark this answers is not in the document]
Comment (SeanCheatham) [reply docs/LocalTestnetSimulatorGuide.md]: Updated

**Review by hwbehrens — COMMENTED** [review]
The git submodule still won't pull, unless you explicitly do `git submodule add -f https://github.com/Topl/protobuf-specs ./topl-grpc/src/main/protobuf`.

I am not sure if that's because the folder already exists in the branch (for legacy reasons I assume) or because of how the sbt file is configured, however.

File: build.sbt [hunk 6]
```diff
@@ -754,14 +754,17 @@ lazy val toplGrpc = project
...
         Process(
-          Seq("git", "checkout", Dependencies.protobufSpecsHash),
-          Some(new File("./topl-grpc/src/main/protobuf")
-          )
+          Seq("git", "submodule", "foreach", "'git pull'")
+        ).! == 0
+      )
+      require(
+        Process(
+          Seq("git", "submodule", "update")
```
[the remark this answers is not in the document]
Comment (hwbehrens) [reply build.sbt:767]: I will correct it, thanks @nandotorterolo!

File: build.sbt [hunk 7]
```diff
@@ -752,8 +752,10 @@ lazy val toplGrpc = project
     scalapbCodeGeneratorOptions += CodeGeneratorOption.FlatPackage,
     updateSubmodulesTask := {
       import sys.process._
-      Process.apply(Seq("git", "submodule", "add", "-f", "git@github.com:Topl/protobuf-specs.git", "./topl-grpc/src/main/protobuf")).!
-      Process.apply(Seq("git", "checkout", Dependencies.protobufSpecsHash), Some(new File("./topl-grpc/src/main/protobuf"))).!
+      require(Process(Seq("git", "submodule", "init")).! == 0)
+      require(Process(Seq("git", "submodule", "foreach", "'git pull'")).! == 0)
```
Comment (hwbehrens) [inline build.sbt:756]: This change works for `sbt compile` but it fails for `sbt "Docker / publishLocal"` so I guess this isn't the correct way to do this. @SeanCheatham Any suggestions on how to approach this from another angle?

**Pull request closed without merging**